### PR TITLE
Tests: add new `InstalledPathsOrderTest`

### PR DESCRIPTION
**Pull request by jrfnl**
## Proposed Changes

This new test class tests that the plugin always registers the `installed_paths` in the same order.

I have confirmed that the tests would fail on various CI builds without the fix from #126.

Notes:
* This test is about Composer and the plugin, so does not need to be tested against multiple PHPCS versions.

This test class covers the following bug previously reported:
* Dealerdirect/phpcodesniffer-composer-installer#125
* Dealerdirect/phpcodesniffer-composer-installer#126

## Related Issues

Related to #92

### CODE REVIEW

File: tests/IntegrationTest/InstalledPathsOrderTest.php
Comment (Potherca) [inline tests/IntegrationTest/InstalledPathsOrderTest.php:109]: Something irks me about the `assertSame` / `preg_match` combination. Not entirely sure what... Could `assertMatchesRegularExpression` be used? :thinking:

**Review by Potherca — DISMISSED** [review]

File: tests/IntegrationTest/InstalledPathsOrderTest.php
Comment (jrfnl) [reply tests/IntegrationTest/InstalledPathsOrderTest.php:109]: Good point. I think I may have created this test before we decided to drop PHP 5.3 support and add the PHPUnit Polyfills. Will fix this up.

Comment (jrfnl) [reply tests/IntegrationTest/InstalledPathsOrderTest.php:109]: Actually, looking at it now, I _do_ see why I didn't use `assertMatchesRegularExpression()`: we need to remember the `$matches` for handling and use in the _next_ assertion and `assertMatchesRegularExpression()` doesn't have a way to give us access to that.

Comment (jrfnl) [reply tests/IntegrationTest/InstalledPathsOrderTest.php]: Rebased & force pushed the fixed commit. The additional `$message` parameters in the last three assertions is the only change.

**Review by Potherca — APPROVED** [review]

**Comment by jrfnl** [comment]
Rebased without changes after the merge of #180 to get a clean and passing build. Will merge after the build passes.

**Pull request merged**
